src/logic/map_objects/tribes/trainingsite.cc

```diff
@@ -484,6 +523,14 @@ void TrainingSite::update_soldier_request(bool did_incorporate) {
 				requesting_weak_trainees_ = false;
 				latest_trainee_was_kickout_ = false;
 			}
+			if (kUpperBoundThreshold_ <= repeated_layoff_ctr_ && 0 == soldiers_.size()) {
```

Comment:
use `empty()` instead of `size() == 0`

Reply:
No problem.. I usually do not write C++, and can forget details like this.